dts/bindings/auxdisplay/zephyr,gpio-7-segment.yaml

```diff
@@ -0,0 +1,88 @@
+# Copyright (c) 2024 Chen Xingyu <hi@xingrz.me>
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  Common GPIO-driven 7-Segment Display
+
+  A 7-segment display is a form of electronic display device for displaying
+  decimal numerals that is an alternative to the more complex dot matrix
+  displays. Seven-segment displays are widely used in digital clocks, electronic
+  meters, basic calculators, and other electronic devices that display numerical
+  information.
+
+  This driver supports 7-segment displays driven by GPIOs, in a widely used
+  circuit design either common anode or common cathode.
+
+  In common anode design, digit-gpios and segment-gpios are configured as active
+  high and active low respectively, meaning that the current flows from
+  digit-gpios to segment-gpios. Vice versa for common cathode.
+
+  Example:
+
+  #include <zephyr/dt-bindings/gpio/gpio.h>
+
+  / {
+          auxdisplay_0: digi-display {
+                  compatible = "zephyr,gpio-7-segment";
+                  columns = <3>;
+                  rows = <1>;
+                  segment-gpios = <&gpio0 0 GPIO_ACTIVE_LOW>, /* A */
+                                  <&gpio0 1 GPIO_ACTIVE_LOW>, /* B */
+                                  <&gpio0 2 GPIO_ACTIVE_LOW>, /* C */
+                                  <&gpio0 3 GPIO_ACTIVE_LOW>, /* D */
+                                  <&gpio0 4 GPIO_ACTIVE_LOW>, /* E */
+                                  <&gpio0 5 GPIO_ACTIVE_LOW>, /* F */
+                                  <&gpio0 6 GPIO_ACTIVE_LOW>, /* G */
+                                  <&gpio0 7 GPIO_ACTIVE_LOW>; /* DP */
+                  digit-gpios = <&gpio1 0 GPIO_ACTIVE_HIGH>,  /* DIG1 */
+                                <&gpio1 1 GPIO_ACTIVE_HIGH>,  /* DIG2 */
+                                <&gpio1 2 GPIO_ACTIVE_HIGH>;  /* DIG3 */
+                  refresh-period-ms = <1>;
+          };
+  };
+
+compatible: "zephyr,gpio-7-segment"
```

Comment:
why put a `zephyr,` prefix if this binding is already based on linux ? why not just keep the linux compatible ?

Reply:
No, it’s not based on Linux, and they are not compatible. This driver was introduced a few months before a similar one was added to the Linux kernel. It’s a completely different implementation, with additional features such as multi-digit support.

Reply:
okay, whether it's based on linux or not, the compatible string is the same `gpio-7-segment` (except for this zephyr, prefix), the one property `segment-gpios` in linux is present here as well in the same way , it doesn't matter what the drivers do, the binding is supposed to be a universal hardware description. Now linux should ideally align to zephyr for the other two properties that this adds on top of what is shared with linux. So please remove the `zephyr,` prefix, this binding is actually good hardware description and `zephyr,` prefix is meant for zephyr-os specific software type things, which this is not